### PR TITLE
feature(simulator): support all extension points in the simulator

### LINKED ISSUE #230: Support all extension points
Now, we only support extension points; Filter, Score and NormalizeScore.
Also https://github.com/kubernetes-sigs/kube-scheduler-simulator/pull/220 will support PostFilter.
I'd suggest that we should support all plugins extension points.

/kind feature
/priority important-longterm
/triage accepted

### PR DESCRIPTION
**Pull request by sanposhiho**
#### What type of PR is this?

/area simulator
/kind feature

#### What this PR does / why we need it:

This PR adds the supports of all extension points (except queuesort and preenqueue).

Note that this PR doesn't change web UI side.

#### Which issue(s) this PR fixes:

Fixes #230 

#### Special notes for your reviewer:

/label tide/merge-method-squash

### CODE REVIEW

File: simulator/export/export.go [hunk 1]
```diff
@@ -1,12 +1,12 @@
 package export
 
-//go:generate mockgen --build_flags=--mod=mod -destination=./mock_$GOPACKAGE/pod.go . PodService
```
Comment (sanposhiho) [inline simulator/export/export.go:3]: Those flags are no longer needed because we don't use vendoring now.

**Review by 196Ikuchil — COMMENTED** [review]
@sanposhiho 
Thank you for your awesome work!
I left some comments. Please check it. 🙏

File: simulator/scheduler/plugin/annotation/annotation.go [hunk 2]
```diff
@@ -1,12 +1,30 @@
...
+	// ReserveResultAnnotationKey has the reserve result.
+	ReserveResultAnnotationKey = "scheduler-simulator/reserve-result"
+	// PermitResultAnnotationKey has the permit result.
+	PermitStatusResultAnnotationKey = "scheduler-simulator/permit-result"
+	// PermitTimeoutResultAnnotationKey has the permit result.
+	PermitTimeoutResultAnnotationKey = "scheduler-simulator/permit-result-timeout"
+	// PreBindResultAnnotationKey has the prebind result.
+	PreBindResultAnnotationKey = "scheduler-simulator/prebind-result"
+	// BindResultAnnotationKey has the prebind result.
+	BindResultAnnotationKey = "scheduler-simulator/bind-result"
```
[the remark this answers is not in the document]
Comment (sanposhiho) [reply simulator/scheduler/plugin/annotation/annotation.go:27]: PostBind doesn't return anything. So, nothing we should record. Unreserve as well.

For QueueSort, it's a bit different extension point and isn't executed in the scheduling cycle/binding cycle.

```
	// Less are used to sort pods in the scheduling queue.
	Less(*QueuedPodInfo, *QueuedPodInfo) bool
```

Also it just contains `less` func to sort heap. So, nothing to record anyway.

File: simulator/scheduler/plugin/plugins.go [hunk 3]
```diff
@@ -168,9 +168,38 @@ func NewPluginConfig(pc []v1beta2.PluginConfig) ([]v1beta2.PluginConfig, error)
 
 // ConvertForSimulator convert v1beta2.Plugins for simulator.
 // It ignores non-default plugin.
+//nolint: cyclop
 func ConvertForSimulator(pls *v1beta2.Plugins) (*v1beta2.Plugins, error) {
 	newpls := pls.DeepCopy()
```
[the remark this answers is not in the document]
Comment (sanposhiho) [reply simulator/scheduler/plugin/plugins.go:174]: No. Wrapplugin won't do anything with `QueueSort`, so no need to do anything with `QueueSort` here.

File: simulator/scheduler/plugin/plugins.go [hunk 4]
```diff
@@ -253,19 +283,70 @@ func mergePluginSet(inTreePluginSet, outOfTreePluginSet v1beta2.PluginSet) v1bet
 	return v1beta2.PluginSet{Enabled: enabledPlugins}
 }
 
-// registeredPlugins returns all registered score plugin and filter plugin.
+// registeredPlugins returns all registered plugins.
+//nolint: funlen,cyclop
 func registeredPlugins() ([]v1beta2.Plugin, error) {
+	var pls []v1beta2.Plugin
+	registeredscorepls, err := config.RegisteredScorePlugins()
```
[the remark this answers is not in the document]
Comment (sanposhiho) [reply simulator/scheduler/plugin/plugins.go:290]: ditto

https://github.com/kubernetes-sigs/kube-scheduler-simulator/pull/266#discussion_r1095353357

**Comment by sanposhiho** [comment]
@196Ikuchil Thanks. Fixed some points based on your reviews. Please retake a look 🙏

**Comment by 196Ikuchil** [comment]
Thanks!

/lgtm

**Pull request merged**